src/test/java/org/takes/rq/RqMultipartTest.java

```diff
@@ -271,6 +272,100 @@ public void parsesHttpBody() throws IOException {
+                        (long) body.getBytes().length
+                    ),
+                    RqMultipartTest.contentDispositionHeader(
+                        "form-data; name=\"name\""
+                    )
+                )
+            ),
+            new RqGreedy(
+                new RqWithHeaders(
+                    new RqFake("", "", body),
+                    RqMultipartTest.contentLengthHeader(0L),
+                    RqMultipartTest.contentDispositionHeader(
+                        "form-data; name=\"content\"; filename=\"a.bin\""
+                    )
+                )
+            )
+        );
+        final RqMultipart.Base multi = new RqMultipart.Base(request);
+        multi.body().close();
+        MatcherAssert.assertThat(
+            multi.part("name").iterator().next(),
+            Matchers.notNullValue()
+        );
+        try {
+            multi.part("name").iterator().next().body().read();
+            Assert.fail("An IOException was expected");
+        } catch (final IOException ex) {
+            MatcherAssert.assertThat(
+                multi.part("content").iterator().next(),
+                Matchers.notNullValue()
+            );
+            multi.part("content").iterator().next().body().read();
+        }
+    }
+
+    /**
+     * RqMultipart.Base can close all parts explicitly even if the request body
+     * has been closed.
+     * @throws IOException If some problem inside
+     */
+    @Test
+    public void closesExplicitlyAllParts() throws IOException {
+        final String body = "RqMultipartTest.closesExplicitlyAllParts";
+        final RqMultipart request = new RqMultipart.Fake(
+            new RqFake(),
+            new RqGreedy(
+                new RqWithHeaders(
+                    new RqFake("", "", body),
+                    RqMultipartTest.contentLengthHeader(
+                        (long) body.getBytes().length
+                    ),
+                    RqMultipartTest.contentDispositionHeader(
+                        "form-data; name=\"foo\""
+                    )
+                )
+            ),
+            new RqGreedy(
+                new RqWithHeaders(
+                    new RqFake("", "", body),
+                    RqMultipartTest.contentLengthHeader(0L),
+                    RqMultipartTest.contentDispositionHeader(
+                        "form-data; name=\"bar\"; filename=\"a.bin\""
+                    )
+                )
+            )
+        );
+        final RqMultipart.Base multi = new RqMultipart.Base(request);
+        multi.body().close();
+        MatcherAssert.assertThat(
+            multi.part("foo").iterator().next(),
+            Matchers.notNullValue()
+        );
+        multi.part("foo").iterator().next().body().close();
+        MatcherAssert.assertThat(
+            multi.part("bar").iterator().next(),
+            Matchers.notNullValue()
+        );
+        multi.part("bar").iterator().next().body().close();
```

Comment:
@mkordas Here the goal is to prove (for backward compatibility reason) that we don't get IOException when we close explicitly a part even after calling `multi.body().close()`. So not getting exception is somehow the assertion here.